File: MyFloatScrollView/src/com/renyu/myfloatscrollview/MainActivity.java
package com.renyu.myfloatscrollview;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.Window;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.renyu.myfloatscrollview.myview.MyScrollView;

public class MainActivity extends ActionBarActivity {
	
	MyScrollView scrollView=null;
	TextView target_view=null;
	ImageView float_view=null;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		requestWindowFeature(Window.FEATURE_NO_TITLE);
		setContentView(R.layout.activity_main);
		
		scrollView=(MyScrollView) findViewById(R.id.scrollView);
		target_view=(TextView) findViewById(R.id.target_view);
		target_view.setOnClickListener(new TextView.OnClickListener() {

			@Override
			public void onClick(View v) {
				// TODO Auto-generated method stub
				Toast.makeText(MainActivity.this, "target", Toast.LENGTH_LONG).show();
			}});
		float_view=(ImageView) findViewById(R.id.float_view);
		float_view.setOnClickListener(new ImageView.OnClickListener() {

			@Override
			public void onClick(View v) {
				// TODO Auto-generated method stub
				Toast.makeText(MainActivity.this, "float", Toast.LENGTH_LONG).show();
			}});
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		// Handle action bar item clicks here. The action bar will
		// automatically handle clicks on the Home/Up button, so long
		// as you specify a parent activity in AndroidManifest.xml.
		int id = item.getItemId();
		if (id == R.id.action_settings) {
			return true;
		}
		return super.onOptionsItemSelected(item);
	}
}

File: LiveDataDemo/app/src/main/java/com/example/administrator/livedatademo/activity/MainActivity.kt
package com.example.administrator.livedatademo.activity

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.databinding.ObservableBoolean
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import com.example.administrator.livedatademo.R
import com.example.administrator.livedatademo.databinding.ActivityMainBinding
import com.example.administrator.livedatademo.model.AutoRefresh
import com.example.administrator.livedatademo.model.Data1
import com.example.administrator.livedatademo.viewmodel.MainViewModel
import com.example.administrator.livedatademo.viewmodel.MainViewModelFactory

class MainActivity : AppCompatActivity() {

    private var vm: MainViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val viewDataBinding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        viewDataBinding.rf = AutoRefresh(ObservableBoolean(false))

        vm = ViewModelProviders.of(this, MainViewModelFactory(viewDataBinding.rf!!)).get(MainViewModel::class.java)
        vm?.responses?.observe(this, Observer<ArrayList<Data1>> {
            Toast.makeText(this, "得到结果", Toast.LENGTH_LONG).show()
            vm?.notifyDataSetChanged(it!!)
        })

        viewDataBinding.adapter = vm?.adapter

        vm?.id?.value = "1"
    }
}
